crates/nu-protocol/src/errors/report_error.rs

```diff
@@ -86,62 +89,75 @@ where
 }
 
 pub fn format_cli_error(
+    config: &Config,
     working_set: &StateWorkingSet,
     error: &dyn miette::Diagnostic,
     default_code: Option<&'static str>,
 ) -> String {
     format!(
         "Error: {:?}",
-        CliError::new(error, working_set, default_code)
+        CliError::new(config, error, working_set, default_code)
     )
 }
 
-pub fn report_shell_error(engine_state: &EngineState, error: &ShellError) {
-    if engine_state.config.display_errors.should_show(error) {
+pub fn report_shell_error(config: &Config, engine_state: &EngineState, error: &ShellError) {
+    if config.display_errors.should_show(error) {
         let working_set = StateWorkingSet::new(engine_state);
-        report_error(&working_set, error, "nu::shell::error")
+        report_error(config, &working_set, error, "nu::shell::error")
     }
 }
 
-pub fn report_shell_warning(engine_state: &EngineState, warning: &ShellWarning) {
+pub fn report_shell_warning(config: &Config, engine_state: &EngineState, warning: &ShellWarning) {
     if should_show_reportable(engine_state, warning) {
         report_warning(
+            config,
             &StateWorkingSet::new(engine_state),
             warning,
             "nu::shell::warning",
         );
     }
 }
 
-pub fn report_parse_error(working_set: &StateWorkingSet, error: &ParseError) {
-    report_error(working_set, error, "nu::parser::error");
+pub fn report_parse_error(config: &Config, working_set: &StateWorkingSet, error: &ParseError) {
+    report_error(config, working_set, error, "nu::parser::error");
 }
 
-pub fn report_parse_warning(working_set: &StateWorkingSet, warning: &ParseWarning) {
+pub fn report_parse_warning(
+    config: &Config,
+    working_set: &StateWorkingSet,
+    warning: &ParseWarning,
+) {
     if should_show_reportable(working_set.permanent(), warning) {
-        report_warning(working_set, warning, "nu::parser::warning");
+        report_warning(config, working_set, warning, "nu::parser::warning");
     }
 }
 
-pub fn report_compile_error(working_set: &StateWorkingSet, error: &CompileError) {
-    report_error(working_set, error, "nu::compile::error");
+pub fn report_compile_error(config: &Config, working_set: &StateWorkingSet, error: &CompileError) {
+    report_error(config, working_set, error, "nu::compile::error");
 }
 
 pub fn report_experimental_option_warning(
+    config: &Config,
     working_set: &StateWorkingSet,
     warning: &dyn miette::Diagnostic,
 ) {
-    report_warning(working_set, warning, "nu::experimental_option::warning");
+    report_warning(
+        config,
+        working_set,
+        warning,
+        "nu::experimental_option::warning",
+    );
 }
 
 fn report_error(
```

Comment:
Is there any reason to change this function to accept a `config`?

Reply:
I think it'd be good to require passing `stack: Option<&Stack>`, it's easy to make user understand the semantic: if you have a stack, pass it, or else pass None.